### ISSUE TITLE
Python 2 regression from 3.7.0 - AttributeError: 'functools.partial' object has no attribute '__module__'

### ISSUE THREAD

**Issue by jakul** [encode/django-rest-framework#5509]
I have a strange issue which has started to show since DRF 3.7. The `json.load` in my runtime is a `partial`, which then causes a crash when I import `from rest_framework.utils import json`. I'm not sure what package is wrapping `json.load` to make it a partial, but it wasn't a problem in DRF 3.6 or before. @rpkilby This seems to be caused by the work you did to add the `utils.json` module.

## Steps to reproduce

```
In [1]: import json

In [2]: from functools import partial

In [3]: setattr(json, 'load', partial(json.load))

In [4]: json.load
Out[4]: <functools.partial at 0x10af90f70>

In [5]: from rest_framework.utils import json
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-88089cf65770> in <module>()
----> 1 from rest_framework.utils import json

/Users/craig/.virtualenvs/tester/lib/python2.7/site-packages/rest_framework/utils/json.py in <module>()
     29
     30
---> 31 @functools.wraps(json.load)
     32 def load(*args, **kwargs):
     33     kwargs.setdefault('parse_constant', strict_constant)

/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/functools.pyc in update_wrapper(wrapper, wrapped, assigned, updated)
     31     """
     32     for attr in assigned:
---> 33         setattr(wrapper, attr, getattr(wrapped, attr))
     34     for attr in updated:
     35         getattr(wrapper, attr).update(getattr(wrapped, attr, {}))

AttributeError: 'functools.partial' object has no attribute '__module__'
```

## Expected behavior
No exception thrown on import.

## Actual behavior
Exception is thrown.

**Comment by carltongibson** [comment]
Hmmm. Immediate thought is why isn’t the test suite hitting this in travis? 

A: Because we’re not wrapping json.load...

I think we’re going to need more info to be able to reproduce this.

**Comment by rpkilby** [comment]
Hi @jakul. Yep, the relevant changes were made in #5265. 

>  I'm not sure what package is wrapping json.load to make it a partial

Independent of the incompatibility here, I'd suggest finding out what package is doing this, as globally overriding the builtin json functions is probably not the best idea. 

In regards to the incompatibility, I'm not sure if it should be fixed. As stated, some other package is behaving poorly here, and this only has an effect in python 2.

**Comment by rpkilby** [comment]
@carltongibson - this is an interaction with a third party package, not something we would have ran into.

**Comment by carltongibson** [comment]
@jakul I'm going to close this for now: I don't think it's a DRF issue. 

If you can follow-up with more info we'd be very happy to look again, and we can re-open if it is something we can/should address. 

Thanks.

**Comment by jakul** [comment]
I've found the cause of the problem. I am using https://github.com/stefankoegl/python-json-patch to provide JSON Patch support in my API (i.e. my `PATCH` endpoints follow [this standard](https://tools.ietf.org/html/rfc6902)), and it is [this line](https://github.com/stefankoegl/python-json-patch/blob/master/jsonpatch.py#L115) which makes the builtin `json.loads` a partial object.

I found 2 solutions:
* update `rest_framework.utils.json` to use `functools32` instead of `functools`
* execute `from rest_framework.utils import json` before any imports from `jsonpatch`

**Comment by carltongibson** [comment]
OK. Good work!

We would not add an external dependency here. (So `functools32` is out.)

I assume you can just do option 2?

**Comment by jakul** [comment]
By chance I'm in the position that I don't *have* to support Python 2 with Django in my app anymore, so that's the route I've chosen to avoid the problem. Thanks for the help.

**Comment by carltongibson** [comment]
That’s a big win. It’s a much better world. 💃🏼🎉❤️👌🏼

**Comment by rpkilby** [comment]
For reference, this is a known issue (stefankoegl/python-json-patch/issues/37)

**Comment by jakul** [comment]
This has been fixed in `jsonpatch` v1.20
